packages/api/src/external/commonwell/document/document-query-sandbox.ts

```diff
@@ -88,39 +91,44 @@ export async function sandboxGetDocRefsAndUpsert({
     requestId,
   });
 
-  for (const entry of convertibleDocs) {
-    const prevDocId = entry.docRef.id;
-    entry.docRef.id = uuidv7();
-    try {
-      await convertCDAToFHIR({
-        patient,
-        document: {
-          id: entry.docRef.id,
-          content: { mimeType: entry.docRef.content?.[0]?.attachment?.contentType },
-        },
-        s3FileName: entry.s3Info.key,
-        s3BucketName: entry.s3Info.bucket,
-        requestId,
-      });
-
-      const contained = entry.docRef.contained ?? [];
-      const containsPatient = contained.filter(c => c.resourceType === "Patient").length > 0;
-      if (!containsPatient) {
-        contained.push({
-          resourceType: "Patient",
-          id: patient.id,
-        });
-      }
-      entry.docRef.subject = {
-        type: "Patient",
-        reference: `Patient/${patient.id}`,
-      };
+  for (const entry of docsWithContent) {
+    const fileTitle = entry.docRef.content?.[0]?.attachment?.title;
+    if (!existingTitles.includes(fileTitle)) {
```

Comment:
Why the check on titles?

Reply:
I wanted to use DocRef IDs, but they get transformed along the way, so can't really use that - so I just wanted to make sure that we dont create duplicate docrefs for the same file.